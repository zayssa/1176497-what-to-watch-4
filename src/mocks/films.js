export const films = [
  {
    id: 1,
    title: `Stone`,
    poster: `img/avatar.jpg`
  },

  {
    id: 10,
    title: `Glass`,
    poster: `img/aviator.jpg`
  },

  {
    id: 11,
    title: `Window`,
    poster: `img/macbeth.jpg`
  },

  {
    id: 100,
    title: `Hunter`,
    poster: `img/johnny-english.jpg`
  },

  {
    id: 101,
    title: `Alvin`,
    poster: `img/mindhunter.jpg`
  },

  {
    id: 110,
    title: `Wedding`,
    poster: `img/orlando.jpg`
  },

  {
    id: 111,
    title: `Love`,
    poster: `img/snatch.jpg`
  },

  {
    id: 1000,
    title: `Dog and cucumber`,
    poster: `img/player-poster.jpg`
  },
];
