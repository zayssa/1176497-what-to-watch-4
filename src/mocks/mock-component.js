
import React from 'react';

const MockComponent = () => {
  return (<div />);
};

export default MockComponent;
