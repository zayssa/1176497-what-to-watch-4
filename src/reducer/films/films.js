import {extend} from "../../utils/extend";
import {adaptFilm, adaptFilmsList} from "../../utils/adapter-film";

const initialState = {
  films: [],
  comments: {}
};

const ActionType = {
  GET_FILMS: `GET_FILMS`,
  TOGGLE_FAVORITE: `TOGGLE_FAVORITE`,
  ADD_COMMENT: `ADD_COMMENT`
};

const ActionCreator = {
  loadFilms: (films) => ({
    type: ActionType.GET_FILMS,
    payload: films
  }),
  toggleFavorite: (film) => ({
    type: ActionType.TOGGLE_FAVORITE,
    payload: film
  }),
  addComment: (filmId, comments) => ({
    type: ActionType.ADD_COMMENT,
    payload: {filmId, comments}
  })
};

const Operation = {
  loadFilms: () => (dispatch, getState, api) => {
    return api.get(`/films`)
      .then((response) => {
        dispatch(ActionCreator.loadFilms(adaptFilmsList(response.data)));
      });
  },
  toggleFavorite: (filmId) => (dispatch, getState, api) => {
    const currentStatus = getState()[`FILMS`].films.find((film) => film.id === filmId).isFavorite;
    return api.post(`/favorite/${filmId}/${currentStatus ? 0 : 1}`)
      .then((response) => {
        dispatch(ActionCreator.toggleFavorite(adaptFilm(response.data)));
      });
  },
  addComment: (filmId, comment) => (dispatch, getState, api) => {
    return api.post(`/comments/${filmId}`, {body: comment})
      .then((response) => {
        dispatch(ActionCreator.addComment(filmId, response.data));
      });
  }
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.GET_FILMS:
      const comments = {};
      action.payload.forEach((film) => {
        comments[film.id] = [];
      });
      return extend(state, {films: action.payload, comments});
    case ActionType.TOGGLE_FAVORITE:
      return extend(state, {films: state.films.map((film) => (
        film.id === action.payload.id ? action.payload : film
      ))});
    case ActionType.ADD_COMMENT:
      return extend(
          state, {
            comments: extend(
                state.comments, {
                  [action.payload.filmId]: action.payload.comments
                }
            )
          }
      );
  }

  return state;
};

export {reducer, ActionType, ActionCreator, Operation};
