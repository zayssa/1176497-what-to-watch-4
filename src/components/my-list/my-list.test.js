import React from "react";
import {Router} from "react-router-dom";
import renderer from "react-test-renderer";
import MyList from "./my-list.jsx";
import history from "../../history";
import {createAPI} from "../../api";
import {films} from "../../mocks/films";

const api = createAPI();

it(`Render MyList`, () => {
  const tree = renderer
    .create(<Router history={history}><MyList
      filmsList={films.slice(0, 5)}
      userInfo={{
        "id": 1,
        "email": `qwe@mail.ru`,
        "name": `123456`,
        "avatar_url": `img/avatar.png`
      }}
      api={api}
    /></Router>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
