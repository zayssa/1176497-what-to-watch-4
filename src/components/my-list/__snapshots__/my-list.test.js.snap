// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MyList 1`] = `
Array [
  <div
    className="visually-hidden"
  >
    <svg
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <symbol
        id="add"
        viewBox="0 0 19 20"
      >
        <title>
          +
        </title>
        <desc>
          Created with Sketch.
        </desc>
        <g
          fill="none"
          fillRule="evenodd"
          id="Page-1"
          stroke="none"
          strokeWidth="1"
        >
          <polygon
            fill="#EEE5B5"
            id="+"
            points="10.777832 11.2880859 10.777832 19.5527344 8.41650391 19.5527344 8.41650391 11.2880859 0.627929688 11.2880859 0.627929688 8.92675781 8.41650391 8.92675781 8.41650391 0.662109375 10.777832 0.662109375 10.777832 8.92675781 18.5664062 8.92675781 18.5664062 11.2880859"
          />
        </g>
      </symbol>
      <symbol
        id="full-screen"
        viewBox="0 0 27 27"
      >
        <path
          clipRule="evenodd"
          d="M23.8571 0H16V3.14286H23.8571V11H27V3.14286V0H23.8571Z"
          fill="#FFF9D9"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M27 23.8571V16H23.8571V23.8571H16V27H23.8571H27L27 23.8571Z"
          fill="#FFF9D9"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M0 3.14286L0 11H3.14286L3.14286 3.14286L11 3.14286V0H3.14286H0L0 3.14286Z"
          fill="#FFF9D9"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M3.14286 27H11V23.8571H3.14286L3.14286 16H0L0 23.8571V27H3.14286Z"
          fill="#FFF9D9"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
      </symbol>
      <symbol
        id="in-list"
        viewBox="0 0 18 14"
      >
        <path
          clipRule="evenodd"
          d="M2.40513 5.35353L6.1818 8.90902L15.5807 0L18 2.80485L6.18935 14L0 8.17346L2.40513 5.35353Z"
          fill="#EEE5B5"
          fillRule="evenodd"
        />
      </symbol>
      <symbol
        id="pause"
        viewBox="0 0 14 21"
      >
        <title>
          Artboard
        </title>
        <desc>
          Created with Sketch.
        </desc>
        <g
          fill="none"
          fillRule="evenodd"
          id="Artboard"
          stroke="none"
          strokeWidth="1"
        >
          <polygon
            fill="#EEE5B5"
            fillRule="nonzero"
            id="Line"
            points="0 -1.11910481e-13 4 -1.11910481e-13 4 21 0 21"
          />
          <polygon
            fill="#EEE5B5"
            fillRule="nonzero"
            id="Line"
            points="10 -1.11910481e-13 14 -1.11910481e-13 14 21 10 21"
          />
        </g>
      </symbol>
    </svg>
  </div>,
  <div
    className="user-page"
  >
    <header
      className="page-header user-page__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <h1
        className="page-title user-page__title"
      >
        My list
      </h1>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="https://4.react.pages.academyimg/avatar.png"
            width="63"
          />
        </div>
      </div>
    </header>
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
        >
          <a
            href="/films/1"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <video
                muted={true}
                poster="img/avatar.jpg"
                src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
                style={
                  Object {
                    "bottom": 0,
                    "height": "auto",
                    "left": 0,
                    "margin": "auto",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": "100%",
                  }
                }
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/1"
              onClick={[Function]}
            >
              Stone
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <a
            href="/films/10"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <video
                muted={true}
                poster="img/aviator.jpg"
                src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
                style={
                  Object {
                    "bottom": 0,
                    "height": "auto",
                    "left": 0,
                    "margin": "auto",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": "100%",
                  }
                }
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/10"
              onClick={[Function]}
            >
              Glass
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <a
            href="/films/11"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <video
                muted={true}
                poster="img/macbeth.jpg"
                src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
                style={
                  Object {
                    "bottom": 0,
                    "height": "auto",
                    "left": 0,
                    "margin": "auto",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": "100%",
                  }
                }
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/11"
              onClick={[Function]}
            >
              Window
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <a
            href="/films/100"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <video
                muted={true}
                poster="img/johnny-english.jpg"
                src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
                style={
                  Object {
                    "bottom": 0,
                    "height": "auto",
                    "left": 0,
                    "margin": "auto",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": "100%",
                  }
                }
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/100"
              onClick={[Function]}
            >
              Hunter
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
